src/bundle/Resources/public/js/scripts/core/base.chart.js

```diff
@@ -0,0 +1,223 @@
+(function(global, doc, eZ, Chart) {
+    class BaseChart {
+        constructor(data) {
+            this.setData(data);
+            this.lang = document.documentElement.lang.replace('_', '-'); // TO DO: Get this config from settings
+        }
+
+        setData(data) {
+            this.datasets = data.datasets;
+            this.labels = data.labels;
+
+            if (data.summary) {
+                this.summary = data.summary;
+            }
+        }
+
+        getSummaryValue() {}
+
+        getSummaryName(value) {
+            return value.name;
+        }
+
+        getScalesXAxesOption() {
+            let XAxes = [];
+
+            if (this.type === 'line') {
+                XAxes = [
+                    {
+                        display: true,
+                        gridLines: {
+                            display: false,
+                        },
+                        ticks: {
+                            maxRotation: 0,
+                            autoSkip: false,
+                            callback: (value, index, values) => {
+                                return index % this.labelsInterval ? null : value;
+                            },
+                        },
+                    },
+                ];
+            }
+
+            return XAxes;
+        }
+
+        getScalesYAxesOption() {
+            let YAxes = [];
+
+            if (this.type === 'line') {
```

Comment:
`this.type` this is never set in this base class and this class definitely should not know what are the values of this properties